### ISSUE TITLE
Pyarmor Failure

### ISSUE THREAD

**Issue by thelenlucas** [mit-ll/spacegym-kspdg#16]
Pyarmor appears to have some issues when running on Windows. Specifically when running the example in `README.md`, the following is logged: 
```====================================================================================== test session starts =======================================================================================
platform win32 -- Python 3.12.3, pytest-8.3.2, pluggy-1.5.0
rootdir: C:\Users\techn\Documents\spacegym-kspdg
configfile: pyproject.toml
collected 0 items / 1 error

============================================================================================= ERRORS =============================================================================================
_____________________________________________________________________ ERROR collecting tests/ksp_ingame_tests/test_pe1_i3.py _____________________________________________________________________
ImportError while importing test module 'C:\Users\techn\Documents\spacegym-kspdg\tests\ksp_ingame_tests\test_pe1_i3.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
..\..\AppData\Local\Programs\Python\Python312\Lib\importlib\__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests\ksp_ingame_tests\test_pe1_i3.py:15: in <module>
    import kspdg.utils.constants as C
src\kspdg\__init__.py:49: in <module>
    from kspdg.private_src.python3_12.kspdg_envs.lbg1.lg3_envs import LBG1_LG3_I1_Env as LBG1_LG3_I1_V1
src\kspdg\private_src\python3_12\kspdg_envs\__init__.py:2: in <module>
    from kspdg.private_src.python3_12.pyarmor_runtime_000000 import __pyarmor__
src\kspdg\private_src\python3_12\pyarmor_runtime_000000\__init__.py:2: in <module>
    from .pyarmor_runtime import __pyarmor__
E   ModuleNotFoundError: No module named 'kspdg.private_src.python3_12.pyarmor_runtime_000000.pyarmor_runtime'
==================================================================================== short test summary info =====================================================================================
ERROR tests/ksp_ingame_tests/test_pe1_i3.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================================================== 1 error in 0.89s ========================================================================================
```

Additionally, the `pytest` tests fail similarly: 
```====================================================================================== test session starts =======================================================================================
platform win32 -- Python 3.12.3, pytest-8.3.2, pluggy-1.5.0
rootdir: C:\Users\techn\Documents\spacegym-kspdg
configfile: pyproject.toml
collected 6 items / 3 errors

============================================================================================= ERRORS =============================================================================================
_____________________________________________________________ ERROR collecting tests/serverless_tests/test_serverless_base_agent.py ______________________________________________________________
ImportError while importing test module 'C:\Users\techn\Documents\spacegym-kspdg\tests\serverless_tests\test_serverless_base_agent.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
..\..\AppData\Local\Programs\Python\Python312\Lib\importlib\__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests\serverless_tests\test_serverless_base_agent.py:5: in <module>
    from kspdg.agent_api.base_agent import KSPDGBaseAgent
src\kspdg\__init__.py:49: in <module>
    from kspdg.private_src.python3_12.kspdg_envs.lbg1.lg3_envs import LBG1_LG3_I1_Env as LBG1_LG3_I1_V1
src\kspdg\private_src\python3_12\kspdg_envs\__init__.py:2: in <module>
    from kspdg.private_src.python3_12.pyarmor_runtime_000000 import __pyarmor__
src\kspdg\private_src\python3_12\pyarmor_runtime_000000\__init__.py:2: in <module>
    from .pyarmor_runtime import __pyarmor__
E   ModuleNotFoundError: No module named 'kspdg.private_src.python3_12.pyarmor_runtime_000000.pyarmor_runtime'
______________________________________________________________ ERROR collecting tests/serverless_tests/test_serverless_pe1_base.py _______________________________________________________________
ImportError while importing test module 'C:\Users\techn\Documents\spacegym-kspdg\tests\serverless_tests\test_serverless_pe1_base.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
..\..\AppData\Local\Programs\Python\Python312\Lib\importlib\__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests\serverless_tests\test_serverless_pe1_base.py:8: in <module>
    import kspdg.utils.constants as C
src\kspdg\__init__.py:49: in <module>
    from kspdg.private_src.python3_12.kspdg_envs.lbg1.lg3_envs import LBG1_LG3_I1_Env as LBG1_LG3_I1_V1
src\kspdg\private_src\python3_12\kspdg_envs\__init__.py:2: in <module>
    from kspdg.private_src.python3_12.pyarmor_runtime_000000 import __pyarmor__
src\kspdg\private_src\python3_12\pyarmor_runtime_000000\__init__.py:2: in <module>
    from .pyarmor_runtime import __pyarmor__
E   ModuleNotFoundError: No module named 'kspdg.private_src.python3_12.pyarmor_runtime_000000.pyarmor_runtime'
________________________________________________________________ ERROR collecting tests/serverless_tests/test_serverless_utils.py ________________________________________________________________
ImportError while importing test module 'C:\Users\techn\Documents\spacegym-kspdg\tests\serverless_tests\test_serverless_utils.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
..\..\AppData\Local\Programs\Python\Python312\Lib\importlib\__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests\serverless_tests\test_serverless_utils.py:8: in <module>
    import kspdg.utils.constants as C
src\kspdg\__init__.py:49: in <module>
    from kspdg.private_src.python3_12.kspdg_envs.lbg1.lg3_envs import LBG1_LG3_I1_Env as LBG1_LG3_I1_V1
src\kspdg\private_src\python3_12\kspdg_envs\__init__.py:2: in <module>
    from kspdg.private_src.python3_12.pyarmor_runtime_000000 import __pyarmor__
src\kspdg\private_src\python3_12\pyarmor_runtime_000000\__init__.py:2: in <module>
    from .pyarmor_runtime import __pyarmor__
E   ModuleNotFoundError: No module named 'kspdg.private_src.python3_12.pyarmor_runtime_000000.pyarmor_runtime'
==================================================================================== short test summary info =====================================================================================
ERROR tests/serverless_tests/test_serverless_base_agent.py
ERROR tests/serverless_tests/test_serverless_pe1_base.py
ERROR tests/serverless_tests/test_serverless_utils.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 3 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================================================= 3 errors in 0.85s ========================================================================================
```

This prevents using the environment entirely, as the failure occurs on importing the module. Installation has been attempted with both `conda` environments, as well as a standard `pip` installation.

**Comment by rallen10** [comment]
Thank you for bringing this to my attention. This may take a little while to debug, I will need to get back to you next week

**Comment by rallen10** [comment]
I've put together a pre-release that may solve this problem, but I haven't been able to test it on Windows, yet. I'll try to validate it tomorrow, but in the meantime, you can try it yourself:

https://github.com/mit-ll/spacegym-kspdg/releases/tag/v0.9.0-alpha

**Comment by Ian-Meredith** [comment]
Just tried doing a serverless test with the prerelease and it's given me some errors. Not sure if it's a matter of not installing the prerelease correctly or if there's interference with the older release, which I tried deleting to the best of my ability. Any help would be greatly appreciated.

```
$ cd spacegym-kspdg
conda activate kspdg
pytest tests/serverless_tests/
bash: cd: spacegym-kspdg: No such file or directory
============================= test session starts =============================
platform win32 -- Python 3.12.5, pytest-8.3.2, pluggy-1.5.0
rootdir: C:\Users\imere\spacegym-kspdg
configfile: pyproject.toml
collected 3 items / 4 errors

=================================== ERRORS ====================================
_________ ERROR collecting tests/serverless_tests/test_jl_solvers.py __________
tests\serverless_tests\test_jl_solvers.py:20: in <module>
    import kspdg
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:90: in <module>
    __lg3_envs_path = get_private_src_module_str("kspdg_envs.lbg1.lg3_envs")
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:83: in get_private_src_module_str
    platform_architecture = get_platform_architecture()
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:75: in get_platform_architecture
    raise RuntimeError(f"Unsupported architecture: {machine}")
E   RuntimeError: Unsupported architecture: AMD64
____ ERROR collecting tests/serverless_tests/test_serverless_base_agent.py ____
tests\serverless_tests\test_serverless_base_agent.py:5: in <module>
    from kspdg.agent_api.base_agent import KSPDGBaseAgent
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:90: in <module>
    __lg3_envs_path = get_private_src_module_str("kspdg_envs.lbg1.lg3_envs")
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:83: in get_private_src_module_str
    platform_architecture = get_platform_architecture()
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:75: in get_platform_architecture
    raise RuntimeError(f"Unsupported architecture: {machine}")
E   RuntimeError: Unsupported architecture: AMD64
_____ ERROR collecting tests/serverless_tests/test_serverless_pe1_base.py _____
tests\serverless_tests\test_serverless_pe1_base.py:8: in <module>
    import kspdg.utils.constants as C
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:90: in <module>
    __lg3_envs_path = get_private_src_module_str("kspdg_envs.lbg1.lg3_envs")
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:83: in get_private_src_module_str
    platform_architecture = get_platform_architecture()
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:75: in get_platform_architecture
    raise RuntimeError(f"Unsupported architecture: {machine}")
E   RuntimeError: Unsupported architecture: AMD64
______ ERROR collecting tests/serverless_tests/test_serverless_utils.py _______
tests\serverless_tests\test_serverless_utils.py:8: in <module>
    import kspdg.utils.constants as C
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:90: in <module>
    __lg3_envs_path = get_private_src_module_str("kspdg_envs.lbg1.lg3_envs")
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:83: in get_private_src_module_str
    platform_architecture = get_platform_architecture()
..\anaconda3\envs\kspdg\Lib\site-packages\kspdg\__init__.py:75: in get_platform_architecture
    raise RuntimeError(f"Unsupported architecture: {machine}")
E   RuntimeError: Unsupported architecture: AMD64
=========================== short test summary info ===========================
ERROR tests/serverless_tests/test_jl_solvers.py - RuntimeError: Unsupported a...
ERROR tests/serverless_tests/test_serverless_base_agent.py - RuntimeError: Un...
ERROR tests/serverless_tests/test_serverless_pe1_base.py - RuntimeError: Unsu...
ERROR tests/serverless_tests/test_serverless_utils.py - RuntimeError: Unsuppo...
!!!!!!!!!!!!!!!!!!! Interrupted: 4 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 4 errors in 1.16s ==============================
(kspdg)

```

**Comment by rallen10** [comment]
@Ian-Meredith I am able to reproduce your error on my Windows machine and should be able to debug it. I will try to push a new pre-release shortly with a fix

**Comment by rallen10** [comment]
This should be fixed in release [v0.9.0](https://github.com/mit-ll/spacegym-kspdg/releases/tag/v0.9.0). Have confirmed that [pytests](https://github.com/mit-ll/spacegym-kspdg?tab=readme-ov-file#verify-installation) and [evaluation example](https://github.com/mit-ll/spacegym-kspdg?tab=readme-ov-file#example-agent-environment-evaluator-for-scitech-challenge) run without error on a Windows x86_64 and Mac w/ Apple Silicon (arm64) (Linux x86_64 and Mac x86_64 is also supported, but not fully tested).

Closing this issue for now. Let me know if the fix does not work for you.

**Comment by thelenlucas** [comment]
Confirmed, this works on Windows x86_64 now.
